Added support for GE MDS devices

SanderBlom opened the pull request:
Added basic detection for GE MDS devices.

All mibs from the vendor are provided so it's possible to add more detection if needed. 


Please give a short description what your pull request is for

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/librenms/librenms?pullRequest=12834) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/librenms/librenms?pullRequest=12834) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/librenms/librenms?pullRequest=12834) it.</sub>

Jellyfrog commented:
Hi
Congratulations on your first PR!

Could you remove the `.mib` from all MIB files?
And it seems your test data isnt containing anything, could you try rerun it?

SanderBlom commented:
> Hi
> Congratulations on your first PR!
> 
> Could you remove the `.mib` from all MIB files?
> And it seems your test data isnt containing anything, could you try rerun it?

I now removed the .MIB from all the mibs :) 

Regarding the test data, it's still not dumping anything else than what's in the file I committed when I rerun it. 

Commands used to generate test data. 
`./scripts/collect-snmp-data.php -h 7 -v mxncl4ene4sn1s1e6dcnn
./discovery.php -h 7
./scripts/collect-snmp-data.php -h 7 -v mxncl4ene4sn1s1e6dcnn
./scripts/save-test-data.php -o gemds -v mxncl4ene4sn1s1e6dcnn
`

Jellyfrog commented:
Hmm okey try running it against the ip/hostname and enable debug output and post it here:
```
./scripts/collect-snmp-data.php -h *IP* -v mxncl4ene4sn1s1e6dcnn -d
./scripts/save-test-data.php -o gemds -v mxncl4ene4sn1s1e6dcnn -d
```

SanderBlom commented:
> Hmm okey try running it against the ip/hostname and enable debug output and post it here:
> 
> ```
> ./scripts/collect-snmp-data.php -h *IP* -v mxncl4ene4sn1s1e6dcnn -d
> ./scripts/save-test-data.php -o gemds -v mxncl4ene4sn1s1e6dcnn -d
> ```

`librenms@librenms:~$ ./scripts/collect-snmp-data.php -h 172.30.36.50 -v mxncl4ene4sn1s1e6dcnn -d
OS: gemds
Module(s): all
Variant: mxncl4ene4sn1s1e6dcnn

Capturing Data:  sysDescr.0SNMP['/usr/bin/snmpget' '-v2c' '-c' 'public' '-OUneb' '-Ih' '-m' 'SNMPv2-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:172.30.36.50:161' 'sysDescr.0']
 sysObjectID.0SNMP['/usr/bin/snmpget' '-v2c' '-c' 'public' '-OUneb' '-Ih' '-m' 'SNMPv2-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:172.30.36.50:161' 'sysObjectID.0']


Updated snmprec data /opt/librenms/tests/snmpsim/gemds_mxncl4ene4sn1s1e6dcnn.snmprec

Verify this file does not contain any private data before submitting!
librenms@librenms:~$ cat /opt/librenms/tests/snmpsim/gemds_mxncl4ene4sn1s1e6dcnn.snmprec
1.3.6.1.2.1.1.1.0|4|GE MDS Orbit SNMP Agent
1.3.6.1.2.1.1.2.0|6|1.3.6.1.4.1.4130.10
librenms@librenms:~$ 
`
`librenms@librenms:~$ ./scripts/save-test-data.php -o gemds -v mxncl4ene4sn1s1e6dcnn -d
Starting snmpsim listening on 127.1.6.1:1161... 
OS: gemds
Module: all
Variant: mxncl4ene4sn1s1e6dcnn

SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Ovq' '-M' '/opt/librenms/mibs' 'udp:127.1.6.1:1161' 'SNMPv2-MIB::sysDescr.0']
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Ovqn' '-M' '/opt/librenms/mibs' 'udp:127.1.6.1:1161' 'SNMPv2-MIB::sysObjectID.0']
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Oqv' '-m' 'SNMPv2-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'sysName.0']
Added device: 31
127.1.6.1 31 gemds 
#### Load disco module core ####
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OUQn' '-m' 'SNMPv2-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'sysName.0' 'sysObjectID.0' 'sysDescr.0']
OS: GE MDS Orbit network Operating System (gemds)


>> Runtime for discovery module 'core': 0.1700 seconds with 527016 bytes
>> SNMP: [6/1.53s] MySQL: [5/0.01s]
#### Unload disco module core ####


#### Load disco module os ####
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Oqv' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'SNMPv2-MIB::sysLocation.0']
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Ovq' '-m' 'SNMPv2-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'sysContact.0']
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OUQ' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'MDS-SYSTEM-MIB::mSysProductConfiguration.0' 'MDS-SYSTEM-MIB::mSysSerialNumberPlatform.0' 'MDS-SYSTEM-MIB::mSysVersion.2']
Location: 
Hardware: 
OS Version: 
OS Features: 
Serial: 

>> Runtime for discovery module 'os': 0.0450 seconds with 80448 bytes
>> SNMP: [4/0.03s] MySQL: [0/0.00s]
#### Unload disco module os ####


#### Load disco module ports ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IF-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ifDescr']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IF-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ifName']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IF-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ifAlias']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IF-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ifType']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IF-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ifOperStatus']


>> Runtime for discovery module 'ports': 0.0820 seconds with 18432 bytes
>> SNMP: [5/0.08s] MySQL: [1/0.00s]
#### Unload disco module ports ####


#### Load disco module ports-stack ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IF-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ifStackStatus']


>> Runtime for discovery module 'ports-stack': 0.0140 seconds with 3640 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload disco module ports-stack ####


#### Load disco module entity-physical ####

Caching OIDs: entPhysicalEntrySNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'ENTITY-MIB:CISCO-ENTITY-VENDORTYPE-OID-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'entPhysicalEntry']


>> Runtime for discovery module 'entity-physical': 0.0280 seconds with 10248 bytes
>> SNMP: [1/0.02s] MySQL: [1/0.00s]
#### Unload disco module entity-physical ####

Module [ entity-state ] disabled globally.


#### Load disco module processors ####
Host Resources: SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'HOST-RESOURCES-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'hrProcessorLoad']
UCD Resources: SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Ovq' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' '.1.3.6.1.4.1.2021.11.11.0']


>> Runtime for discovery module 'processors': 0.0420 seconds with 66048 bytes
>> SNMP: [2/0.03s] MySQL: [2/0.00s]
#### Unload disco module processors ####


#### Load disco module mempools ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUsetX' '-m' 'HOST-RESOURCES-MIB:HOST-RESOURCES-TYPES' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'hrStorageTable']
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUv' '-m' 'HOST-RESOURCES-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'hrMemorySize.0']
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'UCD-SNMP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'memTotalSwap.0' 'memAvailSwap.0' 'memTotalReal.0' 'memAvailReal.0' 'memBuffer.0' 'memCached.0' 'memSysAvail.0']


>> Runtime for discovery module 'mempools': 0.0680 seconds with 168152 bytes
>> SNMP: [2/0.03s] MySQL: [0/0.00s]
#### Unload disco module mempools ####


#### Load disco module cisco-vrf-lite ####

>> Runtime for discovery module 'cisco-vrf-lite': 0.0050 seconds with 3376 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload disco module cisco-vrf-lite ####

Module [ cisco-mac-accounting ] disabled globally.

Module [ cisco-pw ] disabled globally.

Module [ vrf ] disabled globally.

Module [ cisco-cef ] disabled globally.

Module [ cisco-sla ] disabled globally.

Module [ cisco-cbqos ] disabled globally.

Module [ cisco-otv ] disabled globally.

Module [ cisco-qfp ] disabled globally.


#### Load disco module ipv4-addresses ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Osq' '-m' 'IP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ipAdEntIfIndex']
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Oqv' '-m' 'IP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ipAdEntNetMask.']


>> Runtime for discovery module 'ipv4-addresses': 0.0520 seconds with 11312 bytes
>> SNMP: [2/0.05s] MySQL: [1/0.00s]
#### Unload disco module ipv4-addresses ####


#### Load disco module ipv6-addresses ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Osq' '-Ln' '-m' 'IP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ipAddressIfIndex.ipv6']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OsqnU' '-Ln' '-m' 'IPV6-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ipv6AddrPfxLength']


>> Runtime for discovery module 'ipv6-addresses': 0.0410 seconds with 7536 bytes
>> SNMP: [2/0.03s] MySQL: [1/0.00s]
#### Unload disco module ipv6-addresses ####

Module [ route ] disabled globally.


#### Load disco module sensors ####
Caching data: os sensors SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OteQUs' '-Ih' '-m' 'MDS-SYSTEM-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'mSysTemperature']

 ENTITY-SENSOR: Caching OIDs: entPhysicalDescrSNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'ENTITY-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'entPhysicalDescr']

Airflow: 
Current: 
Charge: 
Dbm: 
Fanspeed: 
Frequency: 
Humidity: 
Load: 
Loss: 
Power: 
Power_consumed: 
Power_factor: 
Runtime: 
Signal: 
State: 
Count: 
Temperature: 
Tv_signal: 
Voltage: 
Snr: 
Pressure: 
Cooling: 
Delay: 
Quality_factor: 
Chromatic_dispersion: 
Ber: 
Eer: 
Waterflow: 
Percent: 

>> Runtime for discovery module 'sensors': 0.0760 seconds with 57296 bytes
>> SNMP: [2/0.04s] MySQL: [33/0.02s]
#### Unload disco module sensors ####


#### Load disco module storage ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IBM-AIX-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'aixFsTableEntry']
aix_filesystem : SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'DATA-DOMAIN-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds:/opt/librenms/mibs/datadomain' 'udp:127.1.6.1:1161' 'fileSystemSpaceTable']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'FREENAS-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'datasetTable']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'FREENAS-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'zpoolTable']
hrStorage : SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'UCD-SNMP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'dskTable']


>> Runtime for discovery module 'storage': 0.1010 seconds with 18944 bytes
>> SNMP: [5/0.08s] MySQL: [1/0.00s]
#### Unload disco module storage ####


#### Load disco module hr-device ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'HOST-RESOURCES-MIB:HOST-RESOURCES-TYPES' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'hrDeviceEntry']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'HOST-RESOURCES-MIB:HOST-RESOURCES-TYPES' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'hrProcessorEntry']


>> Runtime for discovery module 'hr-device': 0.0320 seconds with 4912 bytes
>> SNMP: [2/0.03s] MySQL: [1/0.00s]
#### Unload disco module hr-device ####


#### Load disco module discovery-protocols ####
 CISCO-CDP-MIB: SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUsetX' '-m' 'CISCO-CDP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'cdpCache']

 LLDP-MIB: SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUsetX' '-m' 'LLDP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'lldpRemTable']

 OSPF Discovery: 

>> Runtime for discovery module 'discovery-protocols': 0.0580 seconds with 10120 bytes
>> SNMP: [0/0.00s] MySQL: [3/0.00s]
#### Unload disco module discovery-protocols ####


#### Load disco module arp-table ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUsetX' '-m' 'IP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ipNetToPhysicalPhysAddress']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUsetX' '-m' 'IP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ipNetToMediaPhysAddress']


>> Runtime for discovery module 'arp-table': 0.0400 seconds with 6072 bytes
>> SNMP: [0/0.00s] MySQL: [2/0.00s]
#### Unload disco module arp-table ####

Module [ discovery-arp ] disabled globally.

Module [ junose-atm-vp ] disabled globally.


#### Load disco module bgp-peers ####
SNMP['/usr/bin/snmpgetnext' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUsv' '-m' 'BGP4-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'bgpLocalAs']
No BGP on host

>> Runtime for discovery module 'bgp-peers': 0.0310 seconds with 6248 bytes
>> SNMP: [1/0.03s] MySQL: [2/0.00s]
#### Unload disco module bgp-peers ####


#### Load disco module vlans ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'BRIDGE-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'dot1dBasePortIfIndex']
IEEE8021-Q-BRIDGE-MIB VLANs: SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Oqv' '-m' 'IEEE8021-Q-BRIDGE-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'dot1qVlanVersionNumber.0']


>> Runtime for discovery module 'vlans': 0.0570 seconds with 11088 bytes
>> SNMP: [2/0.05s] MySQL: [1/0.00s]
#### Unload disco module vlans ####

Module [ vmware-vminfo ] disabled globally.

Module [ libvirt-vminfo ] disabled globally.

Module [ printer-supplies ] disabled globally.


#### Load disco module ucd-diskio ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'UCD-DISKIO-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'diskIOEntry']


>> Runtime for discovery module 'ucd-diskio': 0.0150 seconds with 4584 bytes
>> SNMP: [1/0.01s] MySQL: [1/0.00s]
#### Unload disco module ucd-diskio ####

Module [ applications ] disabled globally.


#### Load disco module services ####

>> Runtime for discovery module 'services': 0.0040 seconds with 768 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload disco module services ####


#### Load disco module stp ####
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Oqv' '-m' 'RSTP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'dot1dStpProtocolSpecification.0']


>> Runtime for discovery module 'stp': 0.0130 seconds with 5520 bytes
>> SNMP: [1/0.01s] MySQL: [1/0.00s]
#### Unload disco module stp ####


#### Load disco module ntp ####

>> Runtime for discovery module 'ntp': 0.0000 seconds with 280 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload disco module ntp ####

Module [ loadbalancers ] disabled globally.

Module [ mef ] disabled globally.


#### Load disco module wireless ####

>> Runtime for discovery module 'wireless': 0.0440 seconds with 144536 bytes
>> SNMP: [0/0.00s] MySQL: [24/0.01s]
#### Unload disco module wireless ####


#### Load disco module fdb-table ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUsetX' '-m' 'Q-BRIDGE-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'dot1qTpFdbPort']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUsetX' '-m' 'BRIDGE-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'dot1dTpFdbPort']


>> Runtime for discovery module 'fdb-table': 0.0440 seconds with 4424 bytes
>> SNMP: [0/0.00s] MySQL: [2/0.00s]
#### Unload disco module fdb-table ####


#### Load disco module mpls ####

>> Runtime for discovery module 'mpls': 0.0070 seconds with 23952 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload disco module mpls ####

Discovered in 2.195 seconds


Hostname:    127.1.6.1
Device ID:   31
OS:          gemds


Created directory : /opt/librenms/rrd/127.1.6.1/

#### Load poller module core ####
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQnUt' '-m' 'SNMPv2-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'sysUpTime.0' 'sysName.0' 'sysObjectID.0' 'sysDescr.0']
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQnUst' '-m' 'HOST-RESOURCES-MIB:SNMP-FRAMEWORK-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'snmpEngineTime.0' 'hrSystemUptime.0']

>> Runtime for poller module 'core': 0.0507 seconds with 1776 bytes
>> SNMP: [2/0.04s] MySQL: [0/0.00s]
#### Unload poller module core ####


#### Load poller module availability ####
Availability: 
1 day : 100%
7 days : 100%
30 days : 100%
365 days : 100%

>> Runtime for poller module 'availability': 0.0219 seconds with 54464 bytes
>> SNMP: [0/0.00s] MySQL: [9/0.00s]
#### Unload poller module availability ####

Module [ unix-agent ] disabled globally.


#### Load poller module os ####
Generic :(
Location: 
Hardware: 
OS Version: 
OS Features: 
Serial: 

>> Runtime for poller module 'os': 0.0007 seconds with 552 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload poller module os ####


#### Load poller module ipmi ####

>> Runtime for poller module 'ipmi': 0.0060 seconds with 2216 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module ipmi ####


#### Load poller module sensors ####

>> Runtime for poller module 'sensors': 0.0008 seconds with 2144 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module sensors ####


#### Load poller module processors ####

>> Runtime for poller module 'processors': 0.0008 seconds with 1696 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module processors ####


#### Load poller module mempools ####

>> Runtime for poller module 'mempools': 0.0008 seconds with 1000 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload poller module mempools ####


#### Load poller module storage ####

>> Runtime for poller module 'storage': 0.0009 seconds with 1936 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module storage ####


#### Load poller module netstats ####
 ICMPSNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'icmp']
 IPSNMP['/usr/bin/snmpgetnext' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ipForwDatagrams' 'ipInDelivers' 'ipInReceives' 'ipOutRequests' 'ipInDiscards' 'ipOutDiscards' 'ipOutNoRoutes' 'ipReasmReqds' 'ipReasmOKs' 'ipReasmFails' 'ipFragOKs' 'ipFragFails' 'ipFragCreates' 'ipInUnknownProtos' 'ipInHdrErrors' 'ipInAddrErrors']
 IP-FORWARDSNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQv' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'IP-FORWARD-MIB::ipCidrRouteNumber.0']
 SNMPSNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'SNMPv2-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'snmp']
 TCPSNMP['/usr/bin/snmpgetnext' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'TCP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'tcpActiveOpens' 'tcpPassiveOpens' 'tcpAttemptFails' 'tcpEstabResets' 'tcpCurrEstab' 'tcpInSegs' 'tcpOutSegs' 'tcpRetransSegs' 'tcpInErrs' 'tcpOutRsts']
 TCPHCSNMP['/usr/bin/snmpgetnext' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'TCP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'tcpHCInSegs.0' 'tcpHCOutSegs.0']
 UDPSNMP['/usr/bin/snmpgetnext' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'UDP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'udpInDatagrams' 'udpOutDatagrams' 'udpInErrors' 'udpNoPorts']


>> Runtime for poller module 'netstats': 0.1119 seconds with 5280 bytes
>> SNMP: [7/0.11s] MySQL: [0/0.00s]
#### Unload poller module netstats ####


#### Load poller module hr-mib ####
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OUQs' '-m' 'HOST-RESOURCES-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'hrSystemProcesses.0' 'hrSystemNumUsers.0']


>> Runtime for poller module 'hr-mib': 0.0120 seconds with 1704 bytes
>> SNMP: [1/0.01s] MySQL: [0/0.00s]
#### Unload poller module hr-mib ####


#### Load poller module ucd-mib ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'UCD-SNMP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'systemStats']
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'UCD-SNMP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'laLoadInt.1' 'laLoadInt.2' 'laLoadInt.3']

>> Runtime for poller module 'ucd-mib': 0.0243 seconds with 2752 bytes
>> SNMP: [2/0.02s] MySQL: [0/0.00s]
#### Unload poller module ucd-mib ####


#### Load poller module ipSystemStats ####
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ipSystemStats']


>> Runtime for poller module 'ipSystemStats': 0.0176 seconds with 1904 bytes
>> SNMP: [1/0.02s] MySQL: [0/0.00s]
#### Unload poller module ipSystemStats ####


#### Load poller module ports ####
Caching Oids: Full ports polling SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'IF-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ifXEntry']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUst' '-m' 'IF-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'ifEntry']
dot3StatsDuplexStatusSNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'EtherLike-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'dot3StatsDuplexStatus']
SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'Q-BRIDGE-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'dot1qPvid']


>> Runtime for poller module 'ports': 0.0803 seconds with 28728 bytes
>> SNMP: [4/0.07s] MySQL: [1/0.00s]
#### Unload poller module ports ####


#### Load poller module customoid ####

>> Runtime for poller module 'customoid': 0.0011 seconds with 3088 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module customoid ####


#### Load poller module bgp-peers ####

>> Runtime for poller module 'bgp-peers': 0.0044 seconds with 12760 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module bgp-peers ####

Module [ junose-atm-vp ] disabled globally.

Module [ printer-supplies ] disabled globally.


#### Load poller module ucd-diskio ####

>> Runtime for poller module 'ucd-diskio': 0.0009 seconds with 2240 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module ucd-diskio ####

Module [ wifi ] disabled globally.


#### Load poller module wireless ####

>> Runtime for poller module 'wireless': 0.0010 seconds with 1880 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module wireless ####


#### Load poller module ospf ####
 Processes: SNMP['/usr/bin/snmpbulkwalk' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-OQUs' '-m' 'OSPF-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'OSPF-MIB::ospfGeneralGroup']
0
>> Runtime for poller module 'ospf': 0.0275 seconds with 32784 bytes
>> SNMP: [1/0.02s] MySQL: [0/0.00s]
#### Unload poller module ospf ####

Module [ cisco-ipsec-flow-monitor ] disabled globally.

Module [ cisco-remote-access-monitor ] disabled globally.

Module [ cisco-cef ] disabled globally.

Module [ cisco-sla ] disabled globally.

Module [ cisco-mac-accounting ] disabled globally.

Module [ cipsec-tunnels ] disabled globally.

Module [ cisco-ace-loadbalancer ] disabled globally.

Module [ cisco-ace-serverfarms ] disabled globally.

Module [ cisco-asa-firewall ] disabled globally.

Module [ cisco-voice ] disabled globally.

Module [ cisco-cbqos ] disabled globally.

Module [ cisco-otv ] disabled globally.

Module [ cisco-qfp ] disabled globally.

Module [ cisco-vpdn ] disabled globally.

Module [ nac ] disabled globally.

Module [ netscaler-vsvr ] disabled globally.

Module [ aruba-controller ] disabled globally.


#### Load poller module entity-physical ####

>> Runtime for poller module 'entity-physical': 0.0016 seconds with 2520 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module entity-physical ####

Module [ entity-state ] disabled globally.


#### Load poller module applications ####

>> Runtime for poller module 'applications': 0.0032 seconds with 2152 bytes
>> SNMP: [0/0.00s] MySQL: [1/0.00s]
#### Unload poller module applications ####

Module [ mib ] disabled globally.


#### Load poller module stp ####
SNMP['/usr/bin/snmpget' '-v2c' '-c' 'gemds_mxncl4ene4sn1s1e6dcnn' '-Oqv' '-m' 'RSTP-MIB' '-M' '/opt/librenms/mibs:/opt/librenms/mibs/gemds' 'udp:127.1.6.1:1161' 'dot1dStpProtocolSpecification.0']


>> Runtime for poller module 'stp': 0.0171 seconds with 3192 bytes
>> SNMP: [1/0.01s] MySQL: [1/0.00s]
#### Unload poller module stp ####


#### Load poller module ntp ####

>> Runtime for poller module 'ntp': 0.0006 seconds with 368 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload poller module ntp ####

Module [ loadbalancers ] disabled globally.

Module [ mef ] disabled globally.


#### Load poller module mpls ####

>> Runtime for poller module 'mpls': 0.0003 seconds with 160 bytes
>> SNMP: [0/0.00s] MySQL: [0/0.00s]
#### Unload poller module mpls ####

Enabling graphs: ++++

Polled in 1.466 seconds
Saved to /opt/librenms/tests/data/gemds_mxncl4ene4sn1s1e6dcnn.json
Ready for testing!
librenms@librenms:~$ cat /opt/librenms/tests/data/gemds_mxncl4ene4sn1s1e6dcnn.json
{
    "os": {
        "discovery": {
            "devices": [
                {
                    "sysName": "",
                    "sysObjectID": ".1.3.6.1.4.1.4130.10",
                    "sysDescr": "GE MDS Orbit SNMP Agent",
                    "sysContact": null,
                    "version": null,
                    "hardware": null,
                    "features": null,
                    "os": "gemds",
                    "type": "network",
                    "serial": null,
                    "icon": "ge.svg",
                    "location": null
                }
            ]
        },
        "poller": "matches discovery"
    }
}
librenms@librenms:~$ 
`

Jellyfrog commented on includes/definitions/discovery/gemds.yaml:
```diff
@@ -0,0 +1,15 @@
+--- 
```
```suggestion 
```

Jellyfrog commented:
I updated your yaml file, pull the changes and try again

SanderBlom commented:
Just figured out that the VM was running low on space. WIll increase the disk and partition and try again.

SanderBlom commented:
> I updated your yaml file, pull the changes and try again

Still the same output:

```
librenms@librenms:~$ cat /opt/librenms/tests/data/gemds_mxncl4ene4sn1s1e6dcnn.json
{
    "os": {
        "discovery": {
            "devices": [
                {
                    "sysName": "",
                    "sysObjectID": ".1.3.6.1.4.1.4130.10",
                    "sysDescr": "GE MDS Orbit SNMP Agent",
                    "sysContact": null,
                    "version": null,
                    "hardware": null,
                    "features": null,
                    "os": "gemds",
                    "type": "network",
                    "serial": null,
                    "icon": "ge.svg",
                    "location": null
                }
            ]
        },
        "poller": "matches discovery"
    }
}
librenms@librenms:~$ 

```

murrant commented:
collect-snmp-data.php has something broken regarding debug output... please standby for a fix if I can figure it out.

murrant commented:
Fix pending here: #12837

SanderBlom commented:
> Fix pending here: #12837 after that is merged, run daily.sh and try again.

Thx! 
This fixed the issue :)

SanderBlom commented:
> I updated your yaml file, pull the changes and try again

Test data is now added
